### PR TITLE
Implement form save and resume flow

### LINKED ISSUE #644: Implement a custom react hook to maintain the form state

### PR DESCRIPTION
**Pull request by harishmohanraj**
Fixes #644

### CODE REVIEW

File: app/src/client/components/buildPage/UserProperty.tsx [hunk 1]
```diff
@@ -54,8 +70,8 @@ export const UserProperty = memo(({ activeProperty, propertiesSchema, sideNavIte
   };
 
   useEffect(() => {
-    setActiveModel(null);
-  }, [sideNavItemClickCount]);
+    clearStack();
+  }, [sideNavItemClickCount, clearStack]);
```
Comment (davorinrusevljan) [inline app/src/client/components/buildPage/UserProperty.tsx:74]: why dependency on clearStack function?

File: app/src/client/components/buildPage/useFormHandler.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+export function useFormHandler(propertiesSchema: PropertiesSchema, activeProperty: string) {
+  const [stack, setStack] = useState<PropertySchemaParser[]>([]);
+  console.log('stack: ', stack);
+
+  const getCurrentParser = useCallback(() => {
+    return stack.length > 0 ? stack[stack.length - 1] : null;
+  }, [stack]);
+
+  const addNewParserToStack = useCallback((customOptions: CustomInitOptions) => {
```
Comment (davorinrusevljan) [inline app/src/client/components/buildPage/useFormHandler.ts:23]: maybe `pushNewParser`

File: app/src/client/components/buildPage/UserProperty.tsx (hunk 1 above)
Comment (harishmohanraj) [reply app/src/client/components/buildPage/UserProperty.tsx:74]: Yup, we don't need the clearStack function as part of the dependency.

File: app/src/client/components/buildPage/UserProperty.tsx [hunk 3]
```diff
@@ -20,96 +20,127 @@ interface Props {
...
+        flow: Flow.UPDATE_MODEL,
+        userProperties: userProperties,
+      });
+    };
+
+    useEffect(() => {
+      clearStack();
+    }, [sideNavItemClickCount]);
+
+    const flow = parser?.getFlow();
```
Comment (davorinrusevljan) [inline app/src/client/components/buildPage/UserProperty.tsx:89]: In which situations parser can be null or undefined, and why is in that case ok that flow is undefined?

**Review by davorinrusevljan — APPROVED** [review]
Looks ok for me. The back & forward button should be addressed in separate PR.

**Pull request merged**